micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -710,12 +739,18 @@ public Meter removeByPreFilterId(Meter.Id preFilterId) {
     @Incubating(since = "1.1.0")
     @Nullable
     public Meter remove(Meter.Id mappedId) {
-        Meter m = meterMap.get(mappedId);
-
-        if (m != null) {
+        if (meterMap.containsKey(mappedId)) {
             synchronized (meterMapLock) {
-                m = meterMap.remove(mappedId);
-                if (m != null) {
+                final Meter removedMeter = meterMap.remove(mappedId);
+                Iterator<Meter> iterator = preFilterIdToMeterMap.values().iterator();
+                while (iterator.hasNext()) {
+                    Meter next = iterator.next();
+                    if (next.equals(removedMeter)) {
+                        stalePreFilterIds.remove(next.getId());
```

Comment:
Shouldn't we remove this on the key of the value being removed from preFilterId?

The below test would fail,

``` java
@Test
void meterRegistryShouldBeWipedCompletely() {
    registry.config().commonTags("application", "abcservice");
    Counter c1 = registry.counter("counter");
    registry.config().commonTags("common", "tag");
    
    registry.remove(c1.getId());
    assertThat(registry.getMeters()).isEmpty();
    assertThat(registry.preFilterIdToMeterMap).isEmpty();
    assertThat(registry.stalePreFilterIds).isEmpty();
}
```

Reply:
Good catch. Thanks.